src/packages/documents/documents/entity-actions/culture-and-hostnames/repository/culture-and-hostnames.repository.ts

```diff
@@ -0,0 +1,51 @@
+import { UmbDocumentCultureAndHostnamesServerDataSource } from './culture-and-hostnames.server.data.js';
+import type { UmbControllerHost } from '@umbraco-cms/backoffice/controller-api';
+import { UmbBaseController } from '@umbraco-cms/backoffice/class-api';
+import type { UmbNotificationContext } from '@umbraco-cms/backoffice/notification';
+import { UMB_NOTIFICATION_CONTEXT } from '@umbraco-cms/backoffice/notification';
+import type { UmbApi } from '@umbraco-cms/backoffice/extension-api';
+import type { DomainsPresentationModelBaseModel } from '@umbraco-cms/backoffice/backend-api';
+
+export class UmbDocumentCultureAndHostnamesRepository extends UmbBaseController implements UmbApi {
+	#init!: Promise<unknown>;
+
+	#dataSource: UmbDocumentCultureAndHostnamesServerDataSource;
+
+	#notificationContext?: UmbNotificationContext;
+
+	constructor(host: UmbControllerHost) {
+		super(host);
+
+		this.#dataSource = new UmbDocumentCultureAndHostnamesServerDataSource(this);
+
+		this.#init = Promise.all([
+			this.consumeContext(UMB_NOTIFICATION_CONTEXT, (instance) => {
+				this.#notificationContext = instance as UmbNotificationContext;
```

Comment:
Do you know why there is a type-cast (as UmbNotificationContext) here?

Reply:
Probably either VS Code acting up with the types, or I compared it to the old way of doing it. Is not needed, so Ill just remove it again :)